src/network/networkd-varlink.c

```
@@ -0,0 +1,145 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "networkd-varlink.h"
+#include "varlink.h"
+
+typedef struct LookupParameters {
+        int ifindex;
+} LookupParameters;
+
+
+static int vl_get_link(Manager *m, JsonVariant *variant, Link **ret) {
+        static const JsonDispatch dispatch_table[] = {
+                { "ifindex", JSON_VARIANT_INTEGER, json_dispatch_integer, offsetof(LookupParameters, ifindex), 0 },
+                {}
+        };
+        LookupParameters p = {};
+        int r;
+
+        r = json_dispatch(variant, dispatch_table, NULL, 0, &p);
+        if (r < 0)
+                return r;
+
+        if (p.ifindex <= 0)
+                return -EINVAL;
+
+        return link_get(m, p.ifindex, ret);
+}
+
+static int vl_method_renew(Varlink *varlink, JsonVariant *parameters, VarlinkMethodFlags flags, void *userdata) {
+        Manager *m = userdata;
+        Link *link;
+        int r;
+
+        assert(varlink);
+        assert(m);
+
+        r = vl_get_link(m, parameters, &link);
+        if (r < 0)
+                return r;
+
+        if (link->dhcp_client) {
```

should return an error if there is no dhcp client associated with this, no?